ci_tests/system.py

```diff
@@ -111,153 +122,204 @@ def test_execution_with_upload(self):
+            "demoinfo", "post", "save_ddl", params=params, json=json.loads(ddl)
+        )
+        return response.json(), response.status_code
 
     def add_demo_extras(self, demo_id, demo_extras, demo_extras_name):
         """
         Add DemoExtras to the demo
         """
-        files = {'demoextras': demo_extras}
-        params = {'demo_id': demo_id, 'demoextras_name': demo_extras_name}
-        response = self.post('demoinfo', 'add_demoextras', params=params, files=files)
-        return response.json()
+        files = {"demoextras": demo_extras}
+        params = {"demo_id": demo_id, "demoextras_name": demo_extras_name}
+        response = self.post(
+            "demoinfo", "post", "add_demoextras", params=params, files=files
+        )
+        return response.json(), response.status_code
 
     def add_blob(self, demo_id, blob, demo_title, set_name, pos_in_set):
         """
         Add a blob to the demo
         """
-        files = {'blob': blob}
-        params = {'demo_id': demo_id, 'title': demo_title, 'blob_set': set_name, 'pos_set': pos_in_set}
-        response = self.post('blobs', 'add_blob_to_demo', params=params, files=files)
-        return response.json()
+        files = {"blob": blob}
+        params = {
+            "title": demo_title,
+            "blob_set": set_name,
+            "pos_set": pos_in_set,
+        }
+        response = self.post(
+            "blobs", "post", f"demo_blobs/{demo_id}", params=params, files=files
+        )
+        return response.json(), response.status_code
 
     def get_blob_id(self, demo_id, pos_in_set):
         """
         Return the blob id
         """
-        params = {'demo_id': demo_id}
-        response = self.post('blobs', 'get_blobs', params=params)
+        response = self.post("blobs", "get", f"demo_blobs/{demo_id}")
         json_response = response.json()
-        if json_response.get('status') == 'OK':
-            return json_response['sets'][0]['blobs'][str(pos_in_set)]['id']
+        if response.status_code == 200:
+            return json_response[0]["blobs"][str(pos_in_set)]["id"]
 
     def run_demo_with_blobset(self, demo_id, blob, blob_id, blobset_id):
         """
         Runs the demo with the blob given
         """
         width, height = blob.size
-        params = {'demo_id': demo_id,
-                    'origin': 'blobSet',
-                    'params': {},
-                    'crop_info': {'x': 20, 'y': 9.199999999999996,'width': width, 'height': height, 'rotate': 0, 'scaleX': 1, 'scaleY': 1},
-                    'blobs': {'id_blobs': [blob_id]},                
-                    'setId': blobset_id                  
-                }
-        response = self.post('core', 'run', data={"clientData": json.dumps(params)})
+        params = {
+            "demo_id": demo_id,
+            "origin": "blobSet",
+            "params": {},
+            "crop_info": {
+                "x": 20,
+                "y": 9.199999999999996,
```

Comment:
Do we really need "y": 9.199999999999996 here?
Why not simply 9.2?

Reply:
I'm guessing just to test the some error that happened a long time ago, tests pass with both values..

Reply:
Ok, totally fine then.